### ISSUE TITLE
[BUG] RKE2 - Local etcd snapshot restore does not work

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#38338]
**Rancher Server Setup**
- Rancher version: 2.6-head commit id: `df1b202`
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1

**Information about the Cluster**
- Kubernetes version: `v1.23.8+rke2r1`
- Cluster Type (Local/Downstream): Downstream AWS Node driver cluster - 3 etcd/cp and 3 worker nodes

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Admin

**Describe the bug**
[BUG] RKE2 - Local etcd snapshot restore does not work

**To Reproduce**

- Deploy an AWS node driver cluster - RKE2
- Enabled S3 snapshots
- Take a snapshot on the cluster
- It creates s3 and local snapshots
- Restore from the local snapshot

**Result**
Cluster is stuck in `Updating` state with error `Operation etcd restore failed`
Rancher logs:
```
2022/07/19 21:52:39 [INFO] rkecluster fleet-default/sowmya-etcd-restore: re-electing specific init node for etcd snapshot restore
Tue, Jul 19 2022 2:52:39 pm | 2022/07/19 21:52:39 [ERROR] [planner] rkecluster fleet-default/sowmya-etcd-restore: error encountered during plan processing was operation etcd restore failed
2022/07/19 21:52:49 [ERROR] [CAPI] error watching nodes on target cluster: error creating client and cache for remote cluster: error creating dynamic rest mapper for remote cluster "fleet-default/sowmya-etcd-restore": Get "https://k8s/clusters/c-m-t9jxwx88/api?timeout=10s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
Tue, Jul 19 2022 2:52:49 pm | 2022/07/19 21:52:49 [ERROR] [CAPI] Reconciler error: error creating client and cache for remote cluster: error creating dynamic rest mapper for remote cluster "fleet-default/sowmya-etcd-restore": Get "https:///k8s/clusters/c-m-t9jxwx88/api?timeout=10s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

**Expected Result**
Cluster should come up Active after the restore.

**Comment by Oats87** [comment]
When enabling S3 snapshots with Rancher, we put `etcd-s3: true` in the config file. When performing an etcd snapshot restore, this `etcd-s3: true` gets inherited and causes issues when trying to restore from a local snapshot. PR incoming.

**Comment by Oats87** [comment]
Available to test as of commit ID: 447d78a00b1df57f529ec06dd1ba2d074e839e24

**Comment by sowmyav27** [comment]
**On 2.6head commit id: `c47178e`**

- Deployed an AWS Node driver cluster - 1 node all roles, 1 worker node
- Set ETCD snapshot to S3
- After the cluster came up Active, Take a snapshot s1 on the cluster
- Deploy some workloads on the cluster
- Restore to snapshot - s1 `etcd` restore from the local snapshot
- The cluster goes into Updating state and comes back Active
- Verified the workloads are not available on the cluster.
